File: backend/app/core/envelope.py
from typing import Any


class ApiError(Exception):
    """팀 공통 envelope 에러 응답을 만들기 위한 예외.

    라우터/서비스에서 raise하면 main.py의 exception handler가
    {isSuccess: false, code, message, result: null, error} 형태로 변환한다.
    """

    def __init__(self, status_code: int, code: str, message: str):
        self.status_code = status_code
        self.code = code
        self.message = message
        super().__init__(message)


def success(code: str, message: str, result: Any) -> dict:
    return {
        "isSuccess": True,
        "code": code,
        "message": message,
        "result": result,
        "error": None,
    }


def error(code: str, message: str) -> dict:
    return {
        "isSuccess": False,
        "code": code,
        "message": message,
        "result": None,
        "error": {"code": code, "message": message},
    }

File: frontend/src/types/envelope.ts
/**
 * 팀 공통 envelope 응답. dashboard처럼 result에 본문을 담아 보내는 엔드포인트가 씁니다.
 * 실패는 non-2xx로 오므로 성공 응답에서는 result만 꺼내 쓰면 됩니다.
 */
export interface Envelope<T> {
  isSuccess: boolean
  /** 예: "DASH2000" */
  code: string
  message: string
  result: T
  error: { code: string; message: string } | null
}
